### ISSUE TITLE
p-gated augmentations evaluate every row: a skipped row can make the call raise (RandomEqualize/RandomClahe at p=0) and gets a NaN gradient

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4576]
A 2D intensity augmentation with `p < 1` runs `apply_transform` on **every** row of the batch, then picks rows with `torch.where` (`_AugmentationBase.transform_inputs` in `kornia/augmentation/base.py`; the `p == 1` path skips the blend). The rows the gate skips come back with their input values, but they are still evaluated. That has two visible effects, and both occur under natural sampling, with no hand-edited `batch_prob`:

1. **An input check on a skipped row raises for the whole call, even at `p=0.0`.** `RandomEqualize` and `RandomClahe` reject out-of-range input. An image they would never touch still raises.
2. **A skipped row's gradient is not the identity.** Its output equals its input, but the backward pass goes through the discarded branch. Where that branch's derivative is infinite, `torch.where` turns the zero upstream gradient into NaN.

## Repro (main `90650596`, torch 2.14.0, CPU)

```python
import torch, kornia.augmentation as K

x = torch.linspace(0, 1, 1024).reshape(1, 1, 32, 32).repeat(2, 3, 1, 1)
x[1] *= 2                                   # row 1 is outside [0, 1]
for cls in (K.RandomEqualize, K.RandomClahe):
    try:
        cls(p=0.0)(x)
    except RuntimeError as e:
        print(cls.__name__, "p=0.0:", str(e)[:60])

xs = torch.tensor([0.0, 0.25, 1.0]).view(1, 1, 1, 3).repeat(2, 1, 1, 1).requires_grad_()
out = K.RandomGamma((0.5, 0.5), (1.0, 1.0), p=0.0)(xs)
out.sum().backward()
print(torch.equal(out, xs), xs.grad.flatten().tolist())
```

## Observed

```
RandomEqualize p=0.0: equalize expects input values in [0, 1]. Scale the image int
RandomClahe p=0.0: index 271 is out of bounds for dimension 5 with size 256
True [nan, 1.0, 1.0, nan, 1.0, 1.0]
```

With `p=0.5` both classes raise the same way whenever any row is out of range, whatever the gate draws.

## Expected

A row the gate skips has no effect on the call: it cannot make the call raise, and its gradient is the identity, as for `RandomBrightness((1, 1), p=0.0)`, whose gradient is all ones. Possible fixes:
- evaluate the transform only on the selected rows (`to_apply.any()` / index-select), accepting the data-dependent shape on the `p < 1` path;
- or, if the blend must stay for ONNX/compile, feed the skipped rows a safe substitute before the transform (the `torch.where` double-where pattern), so neither the check nor the derivative sees their values.

## Scope

- The blend is shared by every `_AugmentationBase` subclass. The two raising classes and the NaN gradient were measured on the 2D intensity classes. Geometric classes share the path, and #4497 covers their shape-changing case, which is a different defect: there the skipped rows come back transformed.
- Found while reviewing #4566, which states on the `IntensityAugmentationBase2D` page that every sample is computed and pins it in a `test_wart_*` test named after this issue. A fix flips that pin and deletes the sentence.

## Disposition

**n/a — single symbol** (`_AugmentationBase.transform_inputs`)

Posted on behalf of @ducha-aiki by Claude (Opus 5).
